sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -332,4 +333,29 @@ public static String appendQueryParameter(String url, String key, String value)
             ? url + "&" + key + "=" + value
             : url + "?" + key + "=" + value;
     }
+
+    /**
+     * Extracts the byte buffer for upload operations.
+     *
+     * @param data the {@link Flux} of {@link ByteBuffer}, if specified.
+     * @param optionalLength length of data.
+     * @param blockSize the block size (chunk size) to transfer at a time.
+     * @param dataStream the {@link InputStream}, if specified.
+     * @return the updated {@link Flux} of {@link ByteBuffer}.
+     */
+    public static Flux<ByteBuffer> extractByteBuffer(Flux<ByteBuffer> data, Long optionalLength, Long blockSize,
```

Comment:
Could we move this to the implementation utility class that is shared with all the Storage libraries, so it doesn't need to be in the public API.